### PR TITLE
chore: print pubkey on error

### PR DESCRIPTION
**Pull request by lzap**
A little bit more detailed logging. Public keys can be safely logged (private keys are not stored at all in our DB).

### CODE REVIEW

File: internal/jobs/launch_instance_aws.go
Comment (ezr-ondrej) [inline internal/jobs/launch_instance_aws.go:147]:
```suggestion
		logger.Trace().Msgf("Pubkey with fingerprint '%s' not found, importing type %s key '%s'", fingerprint, pubkey.Type, pubkey.Body)
```

Maybe? :)

**Comment by lzap** [comment]
So the problem turns out in the NAME not Fingerprint. I think I can refactor this patch and implement "try again with a different name".

**Comment by ezr-ondrej** [comment]
That would be awesome! 🧡

**Comment by lzap** [comment]
How about this? :-)

File: internal/jobs/launch_instance_aws.go
Comment (avitova) [inline internal/jobs/launch_instance_aws.go:159]: This would be a sequence of: 0, 1, 3, 7, etc. Is that intentional?
```suggestion
				retryCount += 1
```

Comment (lzap) [reply internal/jobs/launch_instance_aws.go:159]: Haha made my day, no obviously this was a typo.

**Comment by lzap** [comment]
Rebased, please test I did not test this.

File: internal/jobs/launch_instance_aws.go
Comment (avitova) [inline internal/jobs/launch_instance_aws.go:161]: I feel like this might get confusing. This says that the retryCount is < 4, but this retry block is actually performed twice. This is just by the fact that we are starting from 1 and also checking for "<".
Just by the looks of it, I would expect the ImportPubkey to be invoked at least three times. Can we change this a bit so it is more clear how many times we actually run this?

**Review by avitova — COMMENTED** [review]
Sorry for late review. I got a bit lost in the readability of the function, so I have few ideas how to make this better. Feel free to not deal with it. But I would probably do something with the retryCount readability. :)

File: internal/jobs/launch_instance_aws.go
Comment (ezr-ondrej) [inline internal/jobs/launch_instance_aws.go:155]: Why do we need a sorcery here? xD
Wouldn't simple for loop with break suffice here?
I'd personally prefer it as more ppl might be more familiar with it and thus having easier time to read it :)

Comment (lzap) [reply internal/jobs/launch_instance_aws.go:155]: Loop introduces a new indentation level, labels are feature of the language and I did some research and one of the main use across the community is for retries.

If you insist tho, I can remove and introduce a loop.

File: internal/jobs/launch_instance_aws.go
Comment (lzap) [reply internal/jobs/launch_instance_aws.go:161]: This was not intentional, you are right. Rebased a change with starting from zero.

File: internal/jobs/launch_instance_aws.go
Comment (lzap) [reply internal/jobs/launch_instance_aws.go:178]: Yep, this code is clunky. Let me refactor it completely.

**Comment by lzap** [comment]
How about this?

Vastly simplified, removed multiple nested levels, still uses goto which everyone have strong opinions on (we can talk about it).

Please test before merge, this can break things up quite a bit.

**Review by avitova — APPROVED** [review]
Nice! So, I have tested all of the flows, and everything seems to be working correctly.

File: internal/jobs/launch_instance_aws.go
Comment (avitova) [inline internal/jobs/launch_instance_aws.go:160]: Just an idea: add a separator between the key name and new suffix. If we had a key named "key1", it suddenly becomes a "key12" if we upload new. But that is nothing major. :thinking:

Comment (lzap) [reply internal/jobs/launch_instance_aws.go:160]: I don’t thing we ever want to retry more than 3-5 times. If you insist I can rebase, this one looks more "human" tho.

**Comment by ezr-ondrej** [comment]
needs rebase after the error rename

File: internal/jobs/launch_instance_aws.go
Comment (avitova) [reply internal/jobs/launch_instance_aws.go:160]: No, I think that the amount of retries is sufficient. I was overthinking the suffix appending. I think this is okay. :+1:

**Comment by lzap** [comment]
Done.

**Review by ezr-ondrej — APPROVED** [review]
Looks sane to me :+1:
@avitova acked the last version if I understand correctly, so lets get it in.

:warning: I'd like to hold off until we figure out the Kafka issue in prod.

**Comment by ezr-ondrej** [comment]
Kafka now works, we will want to play with it more, but got time.
Thanks @lzap and @avitova! Merging :)

**Comment by ezr-ondrej** [comment]
This actually fixed https://issues.redhat.com/browse/HMS-711 :) I'll link it.

**Pull request merged**
